### PR TITLE
Switch from `localStorage` to `chrome.storage.sync`

### PR DESCRIPTION
**Pull request by ArnaudLigny**
Work in progress (issue #39).

To do:
-  [ ] handle default value by key (instead of global object): https://github.com/sindresorhus/github-notifier-chrome/blob/feature-39-chrome.storage.sync/extension/api.js#L46
- [ ] managing data storage migration: https://github.com/sindresorhus/github-notifier-chrome/blob/feature-39-chrome.storage.sync/extension/main.js#L110
- [ ] delete "console.debug" traces

### CODE REVIEW

File: extension/api.js [hunk 1]
```diff
@@ -35,23 +35,48 @@
...
-					var item = localStorage.getItem(name);
-					if (item === null) {
-						return {}.hasOwnProperty.call(defaults, name) ? defaults[name] : undefined;
-					} else if (item === 'true' || item === 'false') {
-						return item === 'true';
-					}
-					return item;
+				get: function (data, callback, sync) {
+					sync = true;
+					var storage = sync ? chrome.storage.sync : chrome.storage.local;
```
Comment (sindresorhus) [inline extension/api.js]: Why are you using both `sync` and `local`? Only `sync`, is good enough.

**Comment by sindresorhus** [comment]
We should subscribe to the [`onInstalled`](https://developer.chrome.com/extensions/runtime#event-onInstalled) event and move the existing data from localStorage to the chrome sync when the user upgrades to the latest version.

File: extension/main.js [hunk 2]
```diff
@@ -17,21 +17,23 @@
...
-
-			if (interval !== null && interval !== intervalSetting) {
-				window.GitHubNotify.settings.set('interval', interval);
-				period = Math.ceil(interval / 60);
-			}
-
-			// unconditionally schedule alarm
-			chrome.alarms.create({when: Date.now() + 2000 + (period * 60 * 1000)});
+			window.GitHubNotify.settings.get('interval', function (items) {
+				intervalSetting = parseInt(items.interval, 10);
```
Comment (sindresorhus) [inline extension/main.js]: We should now save interval as a number, not a string.

File: extension/api.js (hunk 1 above)
Comment (Narno) [inline extension/api.js]: I agree. It was a test.

**Comment by sindresorhus** [comment]
Let me know when this is ready for another review ;)

**Comment by Narno** [comment]
Yep, I have to find some time to work on the to-do list...

**Comment by smblott-github** [comment]
(None of my business, but...)

Would it not be simpler to leave the configuration options in `localStorage` and just use `chrome.storage.sync` for pushing changes between installations?

(Then there'd be no need to change the settings API.  You'd just have a little bit of new code which is isolated from everything else.)

**Comment by Narno** [comment]
@smblott-github it's very clever, I love the idea! o/

**Comment by sindresorhus** [comment]
> (None of my business, but...)

No worries. The beauty of open source is that anyone can comment their opinion.

> Would it not be simpler to leave the configuration options in localStorage and just use chrome.storage.sync for pushing changes between installations?

I don't really see how that's simpler. The migration script is a one time thing and makes the code cleaner in the long term. localStorage is also sub-optimal as it's a key string store without support for types, so we can't store numbers there.

**Comment by smblott-github** [comment]
> I don't really see how that's simpler.

It's simpler because (if I understand the code correctly) you don't need to make any changes to the `settings` API, or to any of the code that uses that API.  So fewer changes, and fewer chances to mess up.

> localStorage is also sub-optimal as it's a key string store without support for types, so we can't store numbers there.

Just use `JSON.stringify` in `settings.set`, and `JSON.parse` in `settings.get`; then you can save any type of data.

**Comment by sindresorhus** [comment]
I guess we have different ways on seeing it. I always prefer the clean code way even if it comes with an additional one time cost.

> So fewer changes, and fewer chances to mess up.

Cleaner code also does that.

**Comment by sindresorhus** [comment]
@Narno ping :)

**Comment by sindresorhus** [comment]
Closing for now. Happy to reopen again if you ever get to this @Narno ;)

**Comment by Narno** [comment]
Yes, I don't have not enough time to contribute currently (new job). Thx.

**Pull request closed without merging**